src/tools/ilasm/ILAssembler/ILAssembler.csproj

```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>
+    <EnableDefaultCompileItems>true</EnableDefaultCompileItems>
+    <Nullable>enable</Nullable>
+    <NoWarn>$(NoWarn);CS3021</NoWarn>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```

Comment:
Does this really need unsafe code?

Reply:
Yes. We have two places that use unsafe code:

1. We use the BlobBuilder constructor, which takes an unmanaged pointer.
2. Native ilasm has the ability to specify an ANSI string that should be interpreted as a Unicode string. The APIs on Marshal to get an ANSI string without using unmanaged memory are all internal, so I'm using the APIs that allocate unmanaged memory. To be explicit, I'm not using `IntPtr`, and the upcoming unsafe definition change would require the code to be marked unsafe anyway.

Reply:
I think you're right. I'll shift it to encode as UTF-8 always.

I'll add the BlobBuilder API to the list of APIs to propose for SRM.

Reply:
Added a `TODO-SRM` in source for this.